### PR TITLE
[GSProcessing] Fix isinstance check for numerical features

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

*Description of changes:*

* Fixes corner case where a feature value might not be one of int/float (we don't cast Parquet input types), replacing with the more general `numbers.Number`.

Example of how to trigger a failure, as observed by a customer:

```python
>>> import numbers
>>> import decimal
>>> dn = decimal.Decimal(5)
>>> intnum = 5
>>> isinstance(intnum, (float, int))
True
>>> isinstance(dn, (float, int))
False
>>> isinstance(dn, numbers.Number)
True
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py [hunk 1]
```diff
@@ -1085,8 +1085,9 @@ def write_processed_feature(
                 }
                 node_type_feature_metadata[feat_name] = node_feature_metadata_dict
 
-                feat_val = single_feature_df.take(1)[0].asDict().get(feat_name, None)
-                nfeat_size = 1 if isinstance(feat_val, (int, float)) else len(feat_val)
```
Comment (jalencato) [inline graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py:1089]: Better to have a comment here to briefly explain about it.

**Review by jalencato — APPROVED** [review]
LGTM, but a small comment.

File: graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py (hunk 1 above)
Comment (thvasilo) [reply graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py:1089]: I'll add a message to the assertion that can act as an explanation as well.

File: graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py [hunk 2]
```diff
@@ -1085,8 +1085,11 @@ def write_processed_feature(
...
+                feat_val = single_feature_df.take(1)[0].asDict().get(feat_name)
+                assert isinstance(
+                    feat_val, (list, numbers.Number)
+                ), "We expect features to either be scalars or lists of scalars."
```
Comment (classicsong) [inline graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py:1091]: If feat_val is a list, shall we check whether every object in the list is a Number?

Comment (thvasilo) [reply graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py:1091]: Sure, I'll add that.

**Comment by thvasilo** [comment]
Made a small change to avoid code repetition and to implement @classicsong suggestion

@jalencato can you take a second look? The last commit pulls out the write_feature function and shares the implementation between nodes and edges.

**Pull request merged**